File: read_rootfile_without_root/read_rootfile_without_root.cc
// name=read_rootfile_without_root ; gcc $name.cc -std=c++11 -lstdc++ -I$HOME/np  -o /tmp/$name && /tmp/$name

#include <iostream>
#include <cstdlib>
#include "NP.hh"

int main(int argc, char** argv)
{
    const char* path = "$JUNOTOP/data/Simulation/SimSvc/PMTSimParamSvc/PMTParam_CD_LPMT.root" ; 
    std::vector<char> buf ; 
    NP::LoadVec<char>(buf, path) ; 

    std::cout << " buf.size " << buf.size() << std::endl ; 

    return 0 ; 
} 

File: read_rootfile_without_root/read_rootfile_without_root.py
#!/usr/bin/env python

import os, numpy as np

if __name__ == '__main__':
    path = os.path.expandvars("$JUNOTOP/data/Simulation/SimSvc/PMTSimParamSvc/PMTParam_CD_LPMT.root")
    a = np.fromfile(path, dtype=np.uint8 )
    print(repr(a))



